frontend/src/features/map_planning/components/BaseLayerConfigurator.tsx

```diff
@@ -0,0 +1,285 @@
+import BaseLayer from '../layers/BaseLayer';
+import { MAP_PIXELS_PER_METER } from '../utils/Constants';
+import { BaseStage } from './BaseStage';
+import { NewBaseLayerDto } from '@/bindings/definitions';
+import SimpleButton from '@/components/Button/SimpleButton';
+import SimpleFormInput from '@/components/Form/SimpleFormInput';
+import ModalContainer from '@/components/Modals/ModalContainer';
+import { KonvaEventObject } from 'konva/lib/Node';
+import { useState } from 'react';
+import { TFunction } from "i18next";
+import { withTranslation } from 'react-i18next';
+import { Layer, Line } from 'react-konva';
+
+export interface BaseLayerConfiguratorProps {
+  onSubmit: (baseLayer: NewBaseLayerDto) => void;
+  t: TFunction;
+}
+
+// Setting the maps scale using a known distance is handled using a state machine.
+enum MeasurementState {
+  Initial, // The user has not selected any points on the map.
+  OnePointSelected, // A single point was selected. Display a line between the selected point and the mouse cursor.
+  TwoPointsSelected, // Two points have been selected. Draw a line between both points.
+}
+
+// Expects an array of line coordinates as returned by Konva [x1, y1, x2, y2].
+export const calculateLineLength = (line: number[]): number => {
+  console.assert(line.length === 4);
+
+  const lineLengthX = Math.abs(line[2] - line[0]);
+  const lineLengthY = Math.abs(line[3] - line[1]);
+
+  return Math.sqrt(lineLengthX * lineLengthX + lineLengthY * lineLengthY);
+};
+
+export const correctForPreviousMapScaling = (distance: number, oldScale: number): number => {
+  return (distance / oldScale) * MAP_PIXELS_PER_METER;
+};
+
+// coordinates of mouse events might be null or undefined
+const mouseEventX = (e: KonvaEventObject<MouseEvent>): number => {
+  const value =
+    e.target.getStage()?.getRelativePointerPosition()?.x == null
+      ? 0
+      : e.target.getStage()?.getRelativePointerPosition()?.x;
+
+  return value ?? 0;
+};
+
+const mouseEventY = (e: KonvaEventObject<MouseEvent>): number => {
+  const value =
+    e.target.getStage()?.getRelativePointerPosition()?.y == null
+      ? 0
+      : e.target.getStage()?.getRelativePointerPosition()?.y;
+
+  return value ?? 0;
+};
+
+const BaseLayerConfigurator = ({ onSubmit, t }: BaseLayerConfiguratorProps) => {
+  const [imageUrl, setImageUrl] = useState('');
+  const [rotation, setImageRotation] = useState(0);
+  const [scale, setImageScale] = useState(10);
+  const [realWorldLength, setRealWorldLength] = useState(0);
+
+  const onUrlInputChange = function <E>(event: React.ChangeEvent<E> | React.KeyboardEvent<E>) {
+    // TODO: add error handling
```

Comment:
Will this be done in this PR?
If not, create an issue addressing this so it will not be forgotten - maybe we can address the error handling in a generalized way

Reply:
Probably not. In my opinion there is no point in adding error messages before Nextcloud is integrated since the code for loading images would need to be rewritten anyway.

I will make sure to create an issue so we don't forget.